database-commons/src/main/java/co/cask/db/batch/action/DBRun.java

```diff
@@ -41,28 +42,24 @@ public DBRun(QueryConfig config, Class<? extends Driver> driverClass) {
    * to use and which connection string to use come from the plugin configuration.
    */
   public void run() throws SQLException, InstantiationException, IllegalAccessException {
-    DBManager dbManager = new DBManager(config);
-
+    DriverCleanup driverCleanup = null;
     try {
-      dbManager.ensureJDBCDriverIsAvailable(driverClass);
+      driverCleanup = DBUtils.ensureJDBCDriverIsAvailable(driverClass, config.getConnectionString(),
+                                                          config.jdbcDriverName);
 
       try (Connection connection = getConnection()) {
-        if (!config.enableAutoCommit) {
```

Comment:
This is still needed for the generic DB action (I think it's a hacky workaround to support the Hive jdbc driver). If it's no longer in the QueryConfig, the boolean should be passed into the constructor.

Reply:
fixed